docs/data/data-grid/overlays/overlays.md

```diff
@@ -0,0 +1,48 @@
+# Data Grid - Overlays
+
+<p class="description">The various data grid overlays.</p>
+
+## Loading overlay
+
+To display a loading overlay and signify that the data grid is in a loading state, set the `loading` prop to `true`.
+
+{{"demo": "LoadingOverlay.js", "bg": "inline"}}
+
+### Variants
+
+The data grid supports 3 loading overlay variants out of the box:
+
+- `circular-progress` (default): a circular loading spinner.
+- `linear-progress`: an indeterminate linear progress bar.
+- `skeleton`: an animated placeholder of the data grid.
+
+The type of loading overlay to display can be set via `slotProps.loadingOverlay` for the following two props:
+
+- `variant`: when `loading` and there are rows in the table.
+- `noRowsVariant`: when `loading` and there are not any rows in the table.
+
+In the following demo, we are showing a skeleton overlay when there are no rows, and a linear progress bar when more are loading:
+
+{{"demo": "LoadingOverlayVariants.js", "bg": "inline"}}
```

Comment:
![image](https://github.com/mui/mui-x/assets/1423607/181407c8-7049-4247-9a50-c6b0135d9bc0)

I see a styling issue for this example, there is a gap after the last column header. It matches the size occupied by the scrollbar so it should be related to that. If you're on macOS the scrollbars are `0` sized so you'll probably see something different. In general for scrollbar issues you can either ask someone else to debug it (I think I'm the only one on linux on the team, everyone else is on macOS), or you can run a linux/windows VM, or you can use browserstack (we have an account, credentials in notion).

Reply:
Managed to reproduce this issue on Mac by setting "Show scroll bars" to "Always".

The issue is apparent with pinned columns on the right as the space is reserved, but the scrollbar is hidden:

<img width="922" alt="Screenshot 2024-06-05 at 15 51 50" src="https://github.com/mui/mui-x/assets/9557798/aaeb643e-dcc8-4664-981d-be0c39a340ec">

Hiding the vertical scrollbar for the skeleton via CSS is problematic. A couple of alternative approaches I'm thinking about:
- Add a `disabled` state to the virtual scrollbar when the skeleton loader is active
- Set `dimensions.hasScrollY` to `false` (somehow) when the skeleton loader is active

Reply:
It would probably be a more robust implementation if the skeleton loader was made of real grid cells instead of an overlay, but I'm not sure if it's worth the time investment. I don't have any strong opinion on those approach, whatever provides the best UX sounds good to me.

Reply:
Found a relatively simple solution: use a scrollbar filler cell to take up the transparent space: https://github.com/mui/mui-x/pull/13293/commits/5567ef3b17f21401533302cec7136cd4333a7a6c